modules/core/src/main/java/org/apache/synapse/mediators/AbstractListMediator.java

```diff
@@ -111,6 +119,14 @@ public boolean mediate(MessageContext synCtx, int mediatorPosition) {
                         returnVal = false;
                         break;
                     }
+                    if (i == mediators.size() - 1) {
```

Comment:
what if the RuntimeStatisticCollector.isStatisticsEnabled() is true? then observers are not notified isn't it?